converter/internal/promtailconvert/internal/build/service_discovery.go

```diff
@@ -107,6 +110,14 @@ func toDiscoveryConfig(cfg *scrapeconfig.Config) prom_discover.Configs {
 	}
 
 	if len(cfg.ServiceDiscoveryConfig.StaticConfigs) != 0 {
+		// Add source and default 'localhost' target to match Promtail's behaviour.
+		// See https://github.com/grafana/loki/blob/4a564456861c0ab7441b815aa49a7c0f22a05f4c/clients/pkg/promtail/targets/file/filetargetmanager.go#L102
```

Comment:
Reflects this behaviour: https://github.com/grafana/loki/blob/4a564456861c0ab7441b815aa49a7c0f22a05f4c/clients/pkg/promtail/targets/file/filetargetmanager.go#L102 

I'm not aware of a better option to do this.